src/transformers/models/swin/modeling_swin.py

```diff
@@ -760,9 +760,17 @@ def forward(
 
 
 class SwinStage(nn.Module):
-    def __init__(self, config, dim, input_resolution, depth, num_heads, drop_path, downsample):
+    def __init__(self, config,grid_size, layer_number, drop_path_rate):
         super().__init__()
         self.config = config
+        dim=int(config.embed_dim * 2**layer_number),
+        
+        input_resolution=(grid_size[0] // (2**layer_number), grid_size[1] // (2**layer_number)),
+        depth=config.depths[layer_number],
+        num_heads=config.num_heads[layer_number],
+        drop_path=drop_path_rate[sum(config.depths[:layer_number]) : sum(config.depths[: layer_number + 1])],
+        downsample=SwinPatchMerging if (layer_number < self.num_layers - 1) else None,
```

Comment:
self.num_layers will be undefined for this class, will do the change required

Reply:
you can pick the number of layers from the `config` object!